File: projects/changestatus-bc/src/main/java/com/robinfood/changestatusbc/mappers/OrderHistoryMappers.java
package com.robinfood.changestatusbc.mappers;

import com.robinfood.changestatusbc.dtos.OrderHistoryDTO;
import com.robinfood.changestatusbc.entities.OrderHistoryEntity;

public final class OrderHistoryMappers {

    private OrderHistoryMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static OrderHistoryEntity toOrderHistoryEntity(OrderHistoryDTO orderHistoryDTO) {
        return OrderHistoryEntity.builder()
                .observation(orderHistoryDTO.getObservation())
                .orderId(orderHistoryDTO.getOrderId())
                .orderStatusId(orderHistoryDTO.getOrderStatusId())
                .userId(orderHistoryDTO.getUserId())
                .build();
    }
}

File: projects/order-bc/app/src/main/java/com/robinfood/app/mappers/input/OrderHistoryMappers.kt
@file:JvmName("OrderHistoryMappers")

package com.robinfood.app.mappers.input

import com.robinfood.core.dtos.request.order.OrderHistoryDTO
import com.robinfood.core.entities.OrderHistoryEntity

fun OrderHistoryDTO.toOrderHistoryEntity(): OrderHistoryEntity {
    return OrderHistoryEntity(
        null,
        null,
        null,
        observation,
        orderId,
        orderStatusId,
        userId
    )
}
